This patch makes libiris svn r982 compile while located in the 
Kopete source tree
Index: iris/xmpp-im/xmpp_task.cpp
===================================================================
--- iris/xmpp-im/xmpp_task.cpp	(revision 769780)
+++ iris/xmpp-im/xmpp_task.cpp	(working copy)
@@ -237,7 +237,7 @@
 		buf = new char[size];
 		va_list ap;
 		va_start(ap, fmt);
-		r = QT_VSNPRINTF(buf, size, fmt, ap);
+		r = qvsnprintf(buf, size, fmt, ap);
 		va_end(ap);
 
 		if(r != -1)
Index: iris/xmpp-core/jid.cpp
===================================================================
--- iris/xmpp-core/jid.cpp	(revision 769780)
+++ iris/xmpp-core/jid.cpp	(working copy)
@@ -23,7 +23,7 @@
 #include <QApplication>
 #include <QByteArray>
 #include <QHash>
-#include <libidn/stringprep.h>
+#include <stringprep.h>
 
 
 using namespace XMPP;
Index: iris/xmpp-core/connector.cpp
===================================================================
--- iris/xmpp-core/connector.cpp	(revision 769780)
+++ iris/xmpp-core/connector.cpp	(working copy)
@@ -36,7 +36,7 @@
 #include <QList>
 #include <QUrl>
 #include "safedelete.h"
-#include <libidn/idna.h>
+#include <idna.h>
 
 #ifdef NO_NDNS
 #include <q3dns.h>
